perfkitbenchmarker/linux_packages/cuda_toolkit_8.py

```diff
@@ -51,21 +70,62 @@ class NvidiaSmiParseOutputException(Exception):
   pass
 
 
+class HeterogeneousGpuTypesException(Exception):
+  pass
+
+
+class UnsupportedGpuTypeException(Exception):
+  pass
+
+
 def GetMetadata(vm):
   """Returns gpu-specific metadata as a dict.
 
   Returns:
     A dict of gpu-specific metadata.
   """
-
   metadata = {}
-  metadata['gpu_memory_clock'] = FLAGS.gpu_clock_speeds[0]
-  metadata['gpu_graphics_clock'] = FLAGS.gpu_clock_speeds[1]
-  metadata['gpu_autoboost_enabled'] = FLAGS.gpu_autoboost_enabled
+  clock_speeds = QueryGpuClockSpeed(vm, 0)
+  autoboost_policy = QueryAutoboostPolicy(vm, 0)
+  metadata['gpu_memory_clock'] = clock_speeds[0]
+  metadata['gpu_graphics_clock'] = clock_speeds[1]
+  metadata['gpu_autoboost'] = autoboost_policy['autoboost']
+  metadata['gpu_autoboost_default'] = autoboost_policy['autoboost_default']
   metadata['nvidia_driver_version'] = GetDriverVersion(vm)
+  metadata['gpu_type'] = GetGpuType(vm)
   return metadata
 
 
+def GetGpuType(vm):
+  """Return the type of NVIDIA gpu(s) installed on the vm.
+
+  Args:
+    vm: virtual machine to query
+
+  Returns:
+    type of gpus installed on the vm as a string
+
+  Raises:
+    NvidiaSmiParseOutputException: if nvidia-smi output cannot be parsed
+    HeterogeneousGpuTypesException: if more than one gpu type is detected
+  """
+  stdout, _ = vm.RemoteCommand('nvidia-smi -L', should_log=True)
+  try:
+    gpu_types = [line.split(' ')[3] for line in stdout.splitlines() if line]
+  except:
+    raise NvidiaSmiParseOutputException('Unable to parse gpu type')
+  if gpu_types[1:] != gpu_types[:-1]:
+    raise HeterogeneousGpuTypesException(
+        'PKB only supports one type of gpu per VM')
+
+  if gpu_types[0].find('P100') != -1:
```

Comment:
Think it might be more readable to use 'in' instead of find - especially because you don't need the index